.github/workflows/system-install.yml

```diff
@@ -0,0 +1,111 @@
+name: System Install
+
+on:
+  pull_request:
+  workflow_dispatch:
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.ref_name }}-${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+env:
+  CARGO_INCREMENTAL: 0
+  CARGO_NET_RETRY: 10
+  CARGO_TERM_COLOR: always
+  RUSTUP_MAX_RETRIES: 10
+
+jobs:
+  install-ubuntu:
+    name: "Install Python on Ubuntu"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.12"
+
+      - name: "Install Rust toolchain"
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: "Build"
+        run: cargo build
+
+      - name: "Print Python path"
+        run: echo $(which python)
+
+      - name: "Validate global Python install"
+        run: python scripts/check_system_python.py --uv ./target/debug/uv
+
+  install-macos:
+    name: "Install Python on macOS"
+    runs-on: macos-14
+    steps:
+      - uses: actions/checkout@v4
+
+      - name: "Install Python"
+        run: brew install python@3.8
+
+      - name: "Install Rust toolchain"
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: "Build"
+        run: cargo build
+
+      - name: "Print Python path"
+        run: echo $(which python3.11)
+
+      - name: "Validate global Python install"
+        run: python3.11 scripts/check_system_python.py --uv ./target/debug/uv
+
+  install-windows:
+    name: "Install Python on Windows"
+    runs-on: windows-latest
+    steps:
+      - uses: actions/checkout@v4
+
+#      - name: "Install Python"
+#        run: choco install python
+
+      - name: "Install Rust toolchain"
+        run: rustup show
+
+      - uses: Swatinem/rust-cache@v2
+
+      - name: "Build"
+        run: cargo build
+
+      - name: "Print Python path"
+        run: echo $(which python)
```

Comment:
I don't think Windows has an `echo` command or a `which` command

Reply:
It has a `where` command

Reply:
It turns out that it doesn't matter on GHA:

![Screenshot 2024-02-28 at 1 29 42 PM](https://github.com/astral-sh/uv/assets/1309177/12b76e62-2af4-463c-8891-c20bbb2988cb)

Reply:
🤯